WPFHospitalEditor/Repository/MapObjectRepository.cs

```diff
@@ -77,10 +77,10 @@ public void saveAll(List<MapObject> entities)
                 serializer.Serialize(jwriter, entities);
             }
         }
-        public MapObject findMapObjectById(int id)
+        public MapObject findMapObjectById(int id, List<MapObject> mapObjects)
         {
-            var allMapObjects = getAllMapObjects().ToList();
-            foreach (MapObject mapObj in allMapObjects)
+            //var allMapObjects = getAllMapObjects().ToList();
+            foreach (MapObject mapObj in mapObjects)
             {
                 if(mapObj.Id == id)
                 {
```

Comment:
The method loses its meaning because it sends a list of objects.  You should provide a list of objects using the repository as you did before

Reply:
Fixed.